jaeger-core/src/main/java/com/uber/jaeger/senders/HttpSender.java

```diff
@@ -24,61 +24,84 @@
 
 import com.uber.jaeger.thriftjava.Batch;
 import com.uber.jaeger.thriftjava.Process;
+import com.uber.jaeger.thriftjava.Span;
 import java.io.IOException;
-import java.net.URI;
-import java.net.URISyntaxException;
 import java.util.List;
-import org.apache.http.HttpResponse;
-import org.apache.http.HttpStatus;
-import org.apache.http.client.HttpClient;
-import org.apache.http.client.methods.HttpPost;
-import org.apache.http.entity.ByteArrayEntity;
-import org.apache.http.impl.client.DefaultHttpClient;
+import okhttp3.HttpUrl;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
 import org.apache.thrift.TException;
 import org.apache.thrift.TSerializer;
 import org.apache.thrift.protocol.TBinaryProtocol.Factory;
 import org.apache.thrift.protocol.TProtocolFactory;
 
 public class HttpSender extends ThriftSender {
+
   private static final String HTTP_COLLECTOR_JAEGER_THRIFT_FORMAT_PARAM = "format=jaeger.thrift";
   private static final TProtocolFactory PROTOCOL_FACTORY = new Factory();
   private static final TSerializer SERIALIZER = new TSerializer(PROTOCOL_FACTORY);
+  private static final int ONE_MB_IN_BYTES = 1048576;
+  private final OkHttpClient httpClient = new OkHttpClient();
+  private final HttpUrl collectorUrl;
 
-  private final HttpClient httpClient = new DefaultHttpClient();
-  private final URI collectorUri;
+  /**
+   * @param endpoint Jaeger REST endpoint consuming jaeger.thrift, e.g
+   * http://localhost:14268/api/traces
+   */
+  public HttpSender(String endpoint) {
+    this(endpoint, ONE_MB_IN_BYTES);
+  }
+
+  /**
+   * @param endpoint Jaeger REST endpoint consuming jaeger.thrift, e.g
+   * http://localhost:14268/api/traces
+   * @param maxPayloadBytes max bytes to serialize as payload
+   */
+  public HttpSender(String endpoint, int maxPayloadBytes) {
+    this(HttpUrl.parse(String.format("%s?%s", endpoint, HTTP_COLLECTOR_JAEGER_THRIFT_FORMAT_PARAM)),
+        maxPayloadBytes);
+  }
 
   /**
-   * @param endpoint Jaeger REST endpoint consuming jaeger.thrift, e.g http://localhost:14268/api/traces
-   * @param maxPacketSize max packet size
+   * @param url Jaeger url accepting jaeger.thrift. This needs to include query params, etc.
+   * @param maxPayloadBytes max bytes to serialize as payload
    */
-  public HttpSender(String endpoint, int maxPacketSize) {
-    super(PROTOCOL_FACTORY, maxPacketSize);
-    this.collectorUri = constructCollectorUri(endpoint);
+  public HttpSender(HttpUrl url, int maxPayloadBytes) {
```

Comment:
any reason for this to be public? Why even be a separate ctor from L63? If we need it we can add it in the future.